guides/blocks/0119-virtual_device_pool.md

```diff
@@ -0,0 +1,241 @@
+  string)
+
+This is an example of a `realms` object:
+
+```
+{
+  realm: "test",
+  jwt: "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MTIyODA4MjIsImFfcGEiOlsiLio6Oi4qIl19.RjqQGV3-ZNbqJ-wj77Sw41CfsWHhlLf9hBZ1Ni-2TQNeKwdpWlmClA1yrtRv41dtVojWjGPLdPMeC7R7yx46VgJlnJXrNE1ffxflAtls5kE3cMF8h6scqbS4Hy7oP_G2g7XS1c9HCwwkiSRzqu6pUY0XgtI0ss35OhvrpkNn0UC4sGS8uAxQauATbfzBdIvkiWPqF-CUrT23sHISlfEv6Y_cnciszJsGTGhVlG1M2hqYHITgnFOivxCgLPQUpqI0GCjkPbfJp9eBClDXJdvIimzRQ74Hv0arBOC6OrCLvztMAwQaxBm47w9sCjkP-aCMN2JqBpmimzFEU8ulXVjxQcgfnAIWrL1cMo4DOhwPvRmyy6Zc9ahRzQdpYvbzbTTfMkPU6u3m-u9CL5JXI8B1l_RriambWsbUeMsOR0WZG9zC98YUwar9bA1o6S7w-3DzxuXDZ4yEe_VdEO8QHb9G_5S1nYXClFXoeL2tRk4Ib9g2HzewC3Zt7pQCX3ksukaPwsxauivQ6m3C_j6LEPiPR4FG2-rJyWFrnAmltYlYs0z7rP0WQjicNIpIRZkB2ovi8gw-PHXFbVlmFfoTx7yl2wWXxLIFDoDieZGdbYcciOERZZ_JzeF7XEWnJvUPmIGhwa5W-UEhAPHXrYjuvHeWTC-FIV5yvjiiSKYjA-DiZSU"
+}
+```
+
+## `interfaces`
+
+The interfaces supported by the target devices. Note that this should be an array of objects, the
+whole interface definition in must be present for every interface. Right now only datastream
+interfaces are supported.
+
+This is an example of an `interfaces` array:
+
+```json
+[
+  {
+    interface_name: "org.astarte-platform.genericsensors.Values",
+    version_major: 1,
+    version_minor: 0,
+    type: "datastream",
+    ownership: "device",
+    description: "Generic sensors sampled data.",
+    mappings: [
+      {
+        endpoint: "/%{sensor_id}/value",
+        type: "double",
+        explicit_timestamp: true,
+        description: "Sampled real value.",
+        doc: "Datastream of sampled real values."
+      }
+    ]
+  }
+]
+```
+
+# Accepted Input Messages
+
+When receiving a message, the `virtual_device_pool` block expects a key with this format:
+
+```
+realm/device_id/interface_name/path
+```
+
+So, for example, an incoming message with key
+`test/EA8lpOrkR8OJ-bp9dSVYQA/org.astarte-platform.genericsensors.Values/mysensor/value` will produce
+a message sent from a device in the `test` realm with Device ID `EA8lpOrkR8OJ-bp9dSVYQA` on the
+`org.astarte-platform.genericsensors.Values` interface with `/mysensor/value` path.
+
+If you're using a static `virtual_device_pool`, the Device ID must be present in the list of target
+devices. If you're using a dynamic `virtual_device_pool`, the device will be registered if needed.
+
+`data` (and its `type`) must be compatible with the type declared in the mapping of the interface,
+so if an interface expects a `double` value but a `string` is received, the message will be
+discarded.
+
+If the interface supports an `explicit_timestamp`, the message `timestamp` is used as explicit
+timestamp.
+
+* `:key`: a unicode string that identifies the stream the message belongs to.
+* `:metadata`: additional message metadata as a map with string key and string value.
+* `:type`: message [data type](#data-and-types) (e.g. integer, real, boolean, etc...).
```

Comment:
I think that `[data type](#data-and-types)` is not a valid link

Reply:
The whole block is actually a leftover, removing it